File: pset6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters / words * 100
    S = sentences / words * 100

    index = round((0.0588 * L) - (0.296 * S) - 15.8)

    if index >= 16:
        print(f"Grade 16+")
    elif index < 1:
        print(f"Before Grade 1")
    else:
        print(f"Grade {index}")


def count_letters(text):
    text_len = len(text)
    n = 0

    for i in range(text_len):
        if text[i].isalpha():
            n += 1

    return n


def count_words(text):
    text_len = len(text)
    n = 1

    for i in range(text_len):
        if text[i] == " ":
            n += 1

    return n


def count_sentences(text):
    text_len = len(text)
    n = 0

    for i in range(text_len):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            n += 1

    return n


main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{

    string input = get_string("Text: ");

    int letters = count_letters(input);
    int words = count_words(input);
    int sentences = count_sentences(input);


    //int S = ((float) sentences / (float) words) * 100;
    //int L = ((float) letters / (float) words) * 100;

    double L = (double) letters / words * 100;
    double S = (double) sentences / words * 100;

    int index = round(((0.0588 * L) - (0.296 * S) - 15.8));

    //printf("%s\n", input);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }


}

int count_letters(string text)
{
    int number_of_letters = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        char upper = toupper(c);
        int n = (int) upper;

        if (upper >= 65 && upper <= 90)
        {
            number_of_letters ++;
        }
    }

    return number_of_letters;
}

int count_words(string text)
{
    int number_of_words = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        int n = (int) c;
        if (n == 32)
        {
            number_of_words++;
        }
    }
    return number_of_words;
}

int count_sentences(string text)
{
    int number_of_sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        int n = (int) c;
        if (n == 33 || n == 46 || n == 63)
        { 
            number_of_sentences++;
        }
    }

    return number_of_sentences;
}